components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -64,20 +74,387 @@ pub struct SplitResult {
+        // this will be used by balance write flow.
+        self.set_peer_stat(parent_stat);
+        let last_compacted_idx = self
+            .storage()
+            .apply_state()
+            .get_truncated_state()
+            .get_index()
+            + 1;
+        self.set_last_compacted_idx(last_compacted_idx);
+
+        let campaigned = self.maybe_campaign(parent_is_leader);
+        if campaigned {
+            self.set_has_ready();
+        }
+
+        if parent_is_leader {
+            self.set_approximate_size(approximate_size);
+            self.set_approximate_keys(approximate_keys);
+            *self.txn_ext().pessimistic_locks.write() = locks;
+            // The new peer is likely to become leader, send a heartbeat immediately to
+            // reduce client query miss.
+            self.heartbeat_pd(store_ctx);
+        }
+
+        let mut tablet = store_ctx
+            .tablet_factory
+            .open_tablet(
+                region_id,
+                Some(RAFT_INIT_LOG_INDEX),
+                OpenOptions::default().set_create(true),
+            )
+            .unwrap();
+        self.tablet_mut().set(tablet);
+
+        meta.tablet_caches.insert(region_id, self.tablet().clone());
+        meta.regions.insert(region_id, region.clone());
+        let not_exist = meta
+            .region_ranges
+            .insert(enc_end_key(&region), region_id)
+            .is_none();
+        assert!(not_exist, "[region {}] should not exist", region_id);
+        meta.readers
+            .insert(region_id, self.generate_read_delegate());
+        meta.region_read_progress
+            .insert(region_id, self.read_progress().clone());
+
+        drop(meta);
+
+        if last_split_region {
+            // todo: check if the last region needs to split again
+        }
+
+        store_ctx.router.force_send(
+            parent_region_id,
+            PeerMsg::RegionSplitMsg(RegionSplitMsg::SplitRegionInitResp(Box::new(
+                SplitRegionInitResp {
+                    parent_epoch,
+                    child_region_id: region_id,
+                    result: true,
+                },
+            ))),
+        );
+
+        if need_schedule_apply_fsm {
+            self.schedule_apply_fsm(store_ctx);
+        }
+    }
+
+    pub fn handle_peer_split_response<T>(
+        &mut self,
+        store_ctx: &mut StoreContext<EK, ER, T>,
+        resp: Box<SplitRegionInitResp>,
+    ) {
+        let SplitRegionInitResp {
+            parent_epoch,
+            child_region_id,
+            result,
+        } = Box::into_inner(resp);
+
+        assert_eq!(
+            parent_epoch,
+            *self
+                .storage()
+                .region_state()
+                .get_region()
+                .get_region_epoch()
+        );
+
+        let mut split_progress = self.split_progress_mut();
+        *split_progress.get_mut(&child_region_id).unwrap() = true;
+
+        if split_progress.values().all(|v| *v) {
```

Comment:
what if there's one new splitted region failed for some reason,  how to recover from that partial status?